src/main/java/at/hannibal2/skyhanni/data/PetAPI.kt

```diff
@@ -35,31 +194,412 @@ object PetAPI {
+        }
+    }
+
+    private fun handleAutopetMessage(string: String, petItem: NEUInternalName): Boolean {
+        autopetHoverPetPattern.matchMatcher(string) {
+            val level = group("level").toInt()
+            val rarity = LorenzRarity.getByColorCode(group("rarity")[0]) ?: LorenzRarity.ULTIMATE
+            val petName = group("pet")
+            val hasSkin = group("skin") != null
+
+            val fakePetLine = "§r§7[Lvl $level] §r${rarity.chatColorCode}$petName${if (hasSkin) "§r${group("skin")}" else ""}"
+
+            val newPet = PetData(
+                petNameToInternalName(petName, rarity),
+                petName,
+                rarity,
+                petItem,
+                level,
+                levelToXP(level, rarity, petName) ?: 0.0,
+                fakePetLine,
+            )
+            fireEvent(newPet)
+            return true
+        }
+        return false
+    }
+
+    private fun handleAutopetItemMessage(string: String): NEUInternalName? {
+        return autopetHoverPetItemPattern.matchMatcher(string) {
+            NEUInternalName.fromItemNameOrNull(group("item"))
+        }
+    }
+
+    @SubscribeEvent
+    fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
+        if (!petMenuPattern.matches(event.inventoryName)) {
+            inPetMenu = false
+            return
+        }
+        val goBackLore = event.inventoryItems[48]?.getLore() ?: emptyList()
+        if (goBackLore.any { forgeBackMenuPattern.matches(it) }) {
+            inPetMenu = false
+            return
+        }
+
+        inPetMenu = true
+    }
+
+    @SubscribeEvent
+    fun onInventoryClose(event: InventoryCloseEvent) {
+        inPetMenu = false
+    }
+
+    @SubscribeEvent
+    fun onSlotClick(event: GuiContainerEvent.SlotClickEvent) {
+        if (!inPetMenu) return
+        if (event.clickTypeEnum != GuiContainerEvent.ClickType.NORMAL) return
+        val category = event.item?.getItemCategoryOrNull() ?: return
+        if (category != ItemCategory.PET) return
+
+        parsePetAsItem(event.item)
+    }
+
+    private fun parsePetAsItem(item: ItemStack) {
+        val lore = item.getLore()
+
+        if (lore.any { petDespawnMenuPattern.matches(it) }) {
+            fireEvent(null)
+            return
+        }
+
+        getPetDataFromItem(item)
+    }
+
+    private fun getPetDataFromItem(item: ItemStack) {
+        val (_, _, rarity, petItem, _, petXP, _) = parsePetNBT(item.extraAttributes)
+        val (internalName, name, _, _, level, _, skin) = parsePetName(item.displayName)
+
+        val newPet = PetData(
+            internalName,
+            name,
+            rarity,
+            petItem,
+            level,
+            petXP,
+            "§r§7[Lvl $level] §r${rarity.chatColorCode}$name${if (skin != "") "§r${skin}" else ""}",
+        )
+        fireEvent(newPet)
+    }
+
+    private fun parsePetNBT(nbt: NBTTagCompound): PetData {
+        val jsonString = nbt.getTag("petInfo").toString()
```

Comment:
instead of string operations, use the correct nbt parse logic. see neu or item utils classes or somewhere